database/service_db.go

```diff
@@ -165,9 +164,6 @@ func (d *LevelDBServiceDB) Save(s *service.Service) error {
 	if s.Hash == "" {
 		return errCannotSaveWithoutHash
 	}
-	if s.Sid == "" {
```

Comment:
Why removing this test?

The db still save the sid:

```
	// save sid-hash pair of service.
	if err := tx.Put([]byte(sidKeyPrefix+s.Sid), []byte(s.Hash), nil); err != nil {
		tx.Discard()
		return err
```

Reply:
> Why removing this test?
Because core dosen't autogenerate (the previous generated sid was invliad domain)

> The db still save the sid:
I pushed check for sid == empty